src/modules/functions/setup-swagger-docs.function.ts

```diff
@@ -0,0 +1,46 @@
+import { INestApplication } from '@nestjs/common';
+import { DocumentBuilder, OpenAPIObject, SwaggerModule } from '@nestjs/swagger';
+import { join } from 'path';
+import { writeJson } from './write-json.function';
+
+export async function setupSwaggerDocs(app: INestApplication, applicationName = 'API') {
+    // Setup Swagger Documents
+    const options = new DocumentBuilder()
+        .setTitle(applicationName)
+        .setDescription(`Documentation for ${applicationName}`)
+        .addBearerAuth()
+        .build();
+    const document = SwaggerModule.createDocument(app, options);
+
+    // Fix operationIds for proper navigation
+    fixOperationIds(document);
+
+    // Write the Json to directory
+    await writeJson(join(process.cwd(), 'openApi.json'), JSON.stringify(document));
```

Comment:
If this is gonna live in nestjs-common, the file/directory it writes to should probably be configurable either through a config or a parameter/option.

Reply:
yeah good idea! I will update